[feat] 내비게이션 바 구현

rafa-e1 opened the pull request:
## 📚 PR 요약
홈 내비게이션 바 UI 변경 및 버튼 추가

## 📝 작업 내용
- 내비게이션 바 숨김 처리 코드 제거 및 내비게이션 바 틴트 컬러 흰색으로 변경
- HomeScene 폴더 추가 및 HomeViewController를 해당 폴더로 이동
- rightBarButtonItem에 검색 모양 아이콘 추가 및 pushViewController 구현
- SafeAreaBrush 오픈소스 추가 및 safe area 영역 keyGreen 색상으로 색칠

## 📌 참고 사항
아래 구조처럼 '뷰 이름 + Scene' 형식의 폴더 구조로 구성하면 좋을 것 같은데 더 나은 의견이 있으시다면 코멘트로 남겨주시면 감사하겠습니다!
```bash
📁 Presentation
 ├── 📄 MainTabBarController
 ├── 📁 HomeScene
 │    ├── 📁 ViewModel
 │    │    └── 📄 HomeViewModel
 │    └── 📁 View
 │         └── 📄 HomeViewController
 └── 📁 CategoryScene
      ├── 📁 ViewModel
      │    └── 📄 CategoryViewModel
      └── 📁 View
           └── 📄 CategoryViewController
```

## 📷 Screenshot
| rightBarButtonItem | backButton |
|--|--|
| ![Simulator Screenshot - iPhone 15 Pro - 2024-07-26 at 16 10 42](https://github.com/user-attachments/assets/17b4209d-0433-4262-b26b-02e6f598e6d3) | ![Simulator Screenshot - iPhone 15 Pro - 2024-07-26 at 16 10 46](https://github.com/user-attachments/assets/acda0859-492a-4641-80a0-4d6baa7df422) |

## 📮 관련 이슈
- Resolved: #7

rafa-e1 committed:
[style] 내비게이션 바 hidden 속성 제거 및 tintColor 수정

- tintColor: black -> white

rafa-e1 committed:
[move] HomeScene 폴더 추가 및 이동

- HomeViewController.swift 이동

rafa-e1 committed:
[feat] 내비게이션 바 UI 및 바 버튼 구현

- rightBarButtonItem

rafa-e1 committed:
[chore] SafeAreaBrush 라이브러리 추가

- SafeArea 영역을 쉽게 색칠할 수 있게 해주는 오픈소스

rafa-e1 committed:
[style] safe area 영역 색 추가

- keyGreen

minnnidev commented on BookTalk/BookTalk/Sources/Presentation/HomeScene/HomeViewController.swift:
```diff
@@ -0,0 +1,60 @@
+
+private extension HomeViewController {
+    
+    func setupUI() {
```
메서드명 관련하여 `BaseVIewController` 제작하여 새로 PR 올렸습니다.
참고하여 코멘트 남겨주세요!

rafa-e1 committed:
[style] 내비게이션 바 및 safe area UI 변경

- Safe Area 배경색: .accentGreen -> 제거
- 내비게이션 바 배걍색: .accentGreen -> 제거
- 내비게이션 바 틴트 컬러: .white -> .black

rafa-e1 committed:
[chore] 라이브러리 제거

- SafeAreaBrush 제거
